src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py

```diff
@@ -450,6 +450,9 @@ def created_flow(pf: PFClient, randstr: Callable[[str], str], variable_recorder)
     """Create a flow for test."""
     flow_display_name = randstr("flow_display_name")
     flow_source = FLOWS_DIR / "simple_hello_world"
+    tool_json_path = f"{flow_source}/{PROMPT_FLOW_DIR_NAME}/{FLOW_TOOLS_JSON}"
```

Comment:
why previously we need this?  is it for component scenario? service need this to generate signature?

Reply:
we don't need this for python flow in scenario other than component. Han has a function which will generate `flow.json` for flex flow only previously and I changed the logic to generate `flow.tools.json` for dag flow also by mistake in a previous PR.

For component scenario, we still need `flow.tools.json` and it can still be generated via `pf flow validate` and this is mentioned in doc of `flow in azure`.

1 possible change we can make is to generate tools json in `flow test` and local `run create` for python flows.